Sources/MongoKitten/CollectionHelpers/Collection+FindAndModify.swift

```diff
@@ -0,0 +1,115 @@
+import NIO
+import MongoClient
+import MongoKittenCore
+
+extension MongoCollection {
+    // MARK: - Builder Functions (Composable/Chained API)
+    
+    /// Modifies and returns a single document.
+    /// - Parameters:
+    ///   - where: The selection criteria for the modification.
+    ///   - update: If passed a document with update operator expressions, performs the specified modification. If passed a replacement document performs a replacement.
+    ///   - remove: Removes the document specified in the query field. Defaults to `false`
+    ///   - returnValue: Wether to return the `original` or `modified` document.
+    public func findAndModify(where query: Document,
```

Comment:
I'd love some overloads for [MongoKittenQuery](https://github.com/OpenKitten/MongoKitten/blob/master/6.0/Sources/MongoKitten/CollectionHelpers/Collection%2BFind.swift#L17). Other than these extra helpers, I have no reason not to merge this.

Reply:
Got em in.